.github/workflows/deploy-prerelease.yml

```
@@ -0,0 +1,224 @@
+            mv ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}/artifact/*.tar.bz2 ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}) || \
+            (echo "${1}: No valid artifact found. Dependency will be installed from the latest release.";\
+            download_latest_release ${1})
+          }
+
+
+          for dependency_repo in ${{env.DEPENDENCY_REPOS}}; do
+            echo "Getting latest \"${dependency_repo}\" dependency ..."
+    
+            # Get sha and number of latest open PR that changes either of the following files:
+            # [src/**, setup.py, pyproject.toml, .conda/**]
+            pr=$(gh pr list --repo ${{ github.repository_owner }}/${dependency_repo} --state open --draft=False \
+            --json headRefOid,files,number,url --jq '.[] | select(.files[].path | 
+            (startswith("src/") or (. == "setup.py") or (. == "pyproject.toml") or (startswith(".conda/"))))' \
+            | head -n 1)
+              
+            # if a PR is found, find the ID of the latest successful workflow run
+            if [[ -n "$pr" ]]; then
+              pr_sha=$(jq -r '.headRefOid' <<< $pr)
+              pr_number=$(jq '.number' <<< $pr)
+              pr_numbers+=($pr_number)
+              pr_url=$(jq -r '.url' <<< $pr)
+              echo "${dependency_repo}: Found PR #${pr_number}."
+              echo "  PR url: $pr_url"
+              echo "  Commit ref: $pr_sha"
+    
+              run=$(gh run list --repo ${{ github.repository_owner }}/${dependency_repo} --json databaseId,headSha,event,status,url \
+              --jq ".[] | select(.event == \"pull_request\" and .status == \"completed\" and .headSha == \"$pr_sha\")"\
+              | head -n 1)
+    
+              # if a successful workflow run is found, download its artifact
+              if [[ -n "$run" ]]; then
+                run_id=$(jq '.databaseId' <<< $run)
+                run_url=$(jq -r '.url' <<< $run)
+                echo "${dependency_repo}: Found successful workflow run for the PR."
+                echo "  Run ID: $run_id"
+                echo "  Run url: $run_url"
+
+                download_run_artifact ${dependency_repo}
+              else
+                echo "${dependency_repo}: No successful workflow run found. Dependency will be installed from the latest release."
+                download_latest_release ${dependency_repo}
+              fi
+            else
+              echo "${dependency_repo}: No useful open PR found. Dependency will be installed from the latest release."
+              download_latest_release ${dependency_repo}
+            fi
+            dependency_names+=($(gh api repos/ACCESS-NRI-TEST/replace_landsurface/contents/pyproject.toml --jq '.content' | base64 --decode | grep -Po "^name\s*=\s*\K(.*)" | tr -d '"'))
+          ls -l ${dependency_repo}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}
+          done
+          echo "dependecy-names=${dependency_names[@]}" >> $GITHUB_OUTPUT
```

Spelling. Note that this usage will have to be updated in this workflow in other places. 

```suggestion
          echo "dependency-names=${dependency_names[@]}" >> $GITHUB_OUTPUT
```

---

Fixed in [bf245e0b7515535862a8bcbd6d90e843da835161](https://github.com/ACCESS-NRI/access-ram-condaenv/pull/9/commits/bf245e0b7515535862a8bcbd6d90e843da835161)